src/core/thread/mle.cpp

```diff
@@ -2726,15 +2726,15 @@ void Mle::ProcessKeySequence(RxInfo &aRxInfo)
     switch (aRxInfo.mClass)
     {
     case RxInfo::kAuthoritativeMessage:
-        Get<KeyManager>().SetCurrentKeySequence(aRxInfo.mKeySequence);
+        Get<KeyManager>().SetCurrentKeySequence(aRxInfo.mKeySequence, KeyManager::kApplyKeySwitchGuard);
```

Comment:
```suggestion
        Get<KeyManager>().SetCurrentKeySequence(aRxInfo.mKeySequence, KeyManager::kForceUpdate);
```

see comments above and spec text - update is forced without checking the key switch guard timer.

Reply:
Thanks @EskoDijk . I saw this text in spec but somehow miss-read it as we need to apply the guard (I was actually thinking to suggest if it make sense to not apply guard timer check on authoritative MLE msgs). Glad that it's already there.

I changed in the new push.